### ISSUE TITLE
tag system for protocols?

### ISSUE THREAD

**Issue by hansvancalster** [inbo/protocolsource#55]
I got an idea based on a question from Emily Veltjen. The current categorization of protocols is rigid (generic, soil, water, ...) and not easily extendable. Sometimes, it is not 100% clear to which category a protocol should belong. For instance "Generic sampling of a biological matrix for molecular research" could go into category "generic" or category "species". 

The INBO tutorials website has this distinction between categories and tags, maybe we can use this for the protocols website too? Tags allow for more flexibility because multiple tags can be associated with one protocol and they need not come from a predefined list. I don't know how easy / difficult it would be to implement something similar for the protocols website.

**Comment by ElsLommelen** [comment]
Honestly I don't know the current protocol workflow very well to judge the implications.  I know functions automatically place protocols in different folders, which won't be necessary anymore, but I don't know how much coding is behind it.  In any case, as this has big implementations for the storage of the protocols, I don't think it is a good idea to postpone it to 'later': it should either be done before starting adding protocols, or we should consider restarting with a new repo when implementing it later on.

**Comment by florisvdh** [comment]
Interesting idea IMHO, although some new challenges arise:

- literally using the term 'tag' for this causes confusion with the ['general' and 'specific' _tags_](https://github.com/inbo/protocolsource/blob/06a53ddcfa83f9204c3d3f560ddef7cdd574c4fc/README.md?plain=1#L60) we will use for released protocols, which effectively are **git tags**.
- you probably want consistency among chosen tags between protocols, perhaps by presetting available options (e.g. soil, species, ...) – effectively making it collapse with the idea of (fixed) 'categories', which themselves would probably be closely related to combination of protocol type (f, a, i, o, p) and theme (0, 1, 2, 3, 4, 5) and perhaps extending it a little.
- it will cause a hybrid situation:
  - since we still must stick with protocol types (f, a, i, o, p) and themes (0, 1, 2, 3, 4, 5) – that approach is already long embedded in the protocol codes of available protocols – such tagging will not avoid the dilemma of having to choose the type and theme that figures in the protocol code (and determining filepaths)
  - the additional 'category' system (avoiding 'tag' here :stuck_out_tongue_winking_eye:) would overlap with the aim of protocol type and theme, and that may cause confusion too, e.g. then best also repeat the actual protocol type/theme as a category of the protocol, potentially next to other ones. So this makes it quite complex (it already [was](https://github.com/inbo/protocolsource/blob/06a53ddcfa83f9204c3d3f560ddef7cdd574c4fc/README.md?plain=1#L99)).
  
  A simplification would be achieved if we would _only_ go with the categories system, and drop the current protocol code approach :smile:; however I assume that is not desired (it's deeply integrated in workflows and tools) and it may present other disadvantages (e.g. protocols get sorted regardless of categories).

FYI @Eveltjen00

**Comment by hansvancalster** [comment]
We didn't pursue this idea further for the reasons @florisvdh gave. So I'm going to close this issue. Feel free to re-open if there is still a need for this.
